src/frontend/screens/ObservationEdit/index.tsx

```diff
@@ -21,14 +20,23 @@ const m = defineMessages({
   },
 });
 
-const ObservationEdit = ({ navigation }: { navigation: NavigationProp }) => {
+const ObservationEdit = ({
+  navigation,
+  route,
+}: NativeRootNavigationProps<"ObservationEdit">) => {
+  const observationId = route.params?.observationId;
+  useSetHeader({
+    headerTitle: !!observationId ? m.editTitle : m.newTitle,
+    headerRight: () => <SaveButton observationId={observationId} />,
+  });
   const handleCategoryPress = useCallback(() => {
-    navigation.navigate({
-      routeName: "CategoryChooser",
-      // Set a key here so we don't navigate back in the stack when creating a
-      // new observation (which starts with the category chooser screen)
-      key: "fromObservationEdit",
-    });
+    navigation.navigate("CategoryChooser");
+    //   {
+    //   routeName: "CategoryChooser",
+    //   // Set a key here so we don't navigate back in the stack when creating a
+    //   // new observation (which starts with the category chooser screen)
+    //   key: "fromObservationEdit",
+    // });
```

Comment:
my guess is that it's probably to account for this: https://reactnavigation.org/docs/navigation-prop#going-back-from-a-specific-screen

if that's the case, maybe keeping it will be safer?

Reply:
I left it in. They key is not being used anywhere... but ill keep it in for now